tools/cldr-apps/js/src/esm/cldrGui.js

```diff
@@ -23,29 +23,99 @@ let dashboardVisible = false;
 
 /**
  * Set up the DOM and start executing Survey Tool as a single page app
+ * @returns Promise
  */
 function run() {
-  if (GUI_DEBUG) {
-    console.log("Hello my name is cldrGui.run");
-    debugParse();
-  }
-  const guiContainer = document.getElementById(runGuiId);
-  if (!guiContainer) {
+  try {
     if (GUI_DEBUG) {
-      console.log(
-        "cldrGui.run doing nothing since '" + runGuiId + "' not found"
-      );
+      console.log("Hello my name is cldrGui.run");
+      debugParse();
     }
-    return;
+    const guiContainer = document.getElementById(runGuiId);
+    if (!guiContainer) {
+      if (GUI_DEBUG) {
+        console.log(
+          "cldrGui.run doing nothing since '" + runGuiId + "' not found"
+        );
+      }
+      return Promise.resolve();
+    }
+    while (guiContainer.firstChild) {
+      guiContainer.removeChild(guiContainer.firstChild);
+    }
+    guiContainer.innerHTML = getBodyHtml();
+    insertHeader();
```

Comment:
doesn't setting innerHTML effectively make removeChild superfluous?

Reply:
didn't seem to work… 